muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/components/guider/body/aside.tsx

```diff
@@ -21,42 +19,55 @@ interface NavigationState {
+            queryString.stringify(Object.assign({}, locationData, { c: "", l: (locationData.l || []).concat(label.id) }), { arrayFormat: 'bracket' });
           return <NavigationElement modifiers="aside-navigation-guider-flag" icon="flag" key={label.id} iconColor={label.color} isActive={isActive} hash={"?" + hash}
-          editableWrapper={LabelUpdateDialog} editableWrapperArgs={{label: label}} isEditable>{label.name}</NavigationElement>
+            editableWrapper={LabelUpdateDialog} editableWrapperArgs={{ label: label }} isEditable>{label.name}</NavigationElement>
         })}
-      </NavigationTopic>: null}
+      </NavigationTopic> : null}
       <NavigationTopic name={this.props.i18n.text.get("plugin.guider.filters.workspaces")}>
-        {this.props.guider.availableFilters.workspaces.map((workspace: GuiderWorkspaceType)=>{
+        {this.props.guider.availableFilters.workspaces.map((workspace: GuiderWorkspaceType) => {
           let isActive = this.props.guider.activeFilters.workspaceFilters.includes(workspace.id);
           let hash = isActive ?
-            queryString.stringify(Object.assign({}, locationData, {c: "", w: (locationData.w || []).filter((w:string)=>parseInt(w)!==workspace.id)}), {arrayFormat: 'bracket'}) :
-            queryString.stringify(Object.assign({}, locationData, {c: "", w: (locationData.w || []).concat(workspace.id)}), {arrayFormat: 'bracket'});
+            queryString.stringify(Object.assign({}, locationData, { c: "", w: (locationData.w || []).filter((w: string) => parseInt(w) !== workspace.id) }), { arrayFormat: 'bracket' }) :
+            queryString.stringify(Object.assign({}, locationData, { c: "", w: (locationData.w || []).concat(workspace.id) }), { arrayFormat: 'bracket' });
           return <NavigationElement modifiers="aside-navigation-guider-course" icon="books" key={workspace.id} isActive={isActive} hash={"?" + hash}>
             {workspace.name + (workspace.nameExtension ? " (" + workspace.nameExtension + ")" : "")}
           </NavigationElement>
         })}
       </NavigationTopic>
+      <NavigationTopic
+        name={this.props.i18n.text.get("plugin.guider.filters.userGroups")}
+      >
+        {this.props.guider.availableFilters.userGroups.map((userGroup: UserGroupType) => {
+          let isActive = this.props.guider.activeFilters.userGroupFilters.includes(userGroup.id);
+          let hash = isActive ?
+            queryString.stringify(Object.assign({}, locationData, { c: "", u: (locationData.u || []).filter((u: string) => parseInt(u) !== userGroup.id) }), { arrayFormat: 'bracket' }) :
+            queryString.stringify(Object.assign({}, locationData, { c: "", u: (locationData.u || []).concat(userGroup.id) }), { arrayFormat: 'bracket' });
+          return <NavigationElement modifiers="aside-navigation-guider-course" icon="users" key={userGroup.id} isActive={isActive} hash={"?" + hash}>
```

Comment:
I think we should rename this modifier to `aside-navigation-guider-usergroup` so it would follow the same logic as flags also has their own modifier. Also necessary .scss changes are required obviously.